find.go

```diff
@@ -90,14 +94,56 @@ func (s *server) doFind(ctx context.Context, method, source string, req *url.URL
 			return nil, err
 		}
 
+		// checking whether it *might* be double hashed query, i.e.
+		// - either decoded multihash from the path has DBL_SHA2_256 codec
+		// - or it's a POST batch reqest (those do not get submitted often)
+		mightBeDhash := false
+		if req.Method == "GET" {
+			ss := strings.Split(b.Path, "/")
+			// multihash is the last part in the path
+			mh, err := multihash.FromB58String(ss[len(ss)-1])
+			if err == nil {
+				dmh, err := multihash.Decode(mh)
+				mightBeDhash = err == nil && dmh.Code == multihash.DBL_SHA2_256
+			}
+		} else {
+			mightBeDhash = true
+		}
+
 		switch resp.StatusCode {
 		case http.StatusOK:
 			atomic.AddInt32(&count, 1)
 			providers, err := model.UnmarshalFindResponse(data)
-			if err != nil {
-				return nil, err
+			// if there was no error then it was a regular find response
+			if err == nil {
+				return &providers, nil
+			} else {
```

Comment:
In go JSON unmarshalling will ignore unknown keys by default. This means you shouldn't get an error here if the response has `EncryptedProviderResults` instead of the usual `ProviderResults`.

Looking at `FindResponse`, it already has an `EncMultihashResults`; this means if there was anything with that key in the body it gets populated. Which we should be able to directly return.

Reply:
Yes, we just need to align dhstore response struct with `FindResponse`.